src/app/search/search.component.ts

```diff
@@ -31,29 +59,115 @@ export class SearchComponent implements OnInit, OnDestroy {
 
   set searchDetails(value: string) {
     this._searchDetails = value;
-    console.log('search: ', this._searchDetails);
+  }
+
+  set selectedCategory(value: string) {
+    this.resetFilterValues();
+    this._selectedCategory = value;
+
+    this.filteredProducts = findItemsByCategory(this._selectedCategory, this.allProductsBySearch);
+    this.visibleProducts = this.filteredProducts.slice(0, 12);
+  }
+
+  set selectedStatus(value: string) {
+    this.resetFilterValues();
+    this._selectedStatus = value;
+
+    this.filteredProducts = findItemsByStatus(this._selectedStatus, this.allProductsBySearch);
+    this.visibleProducts = this.filteredProducts.slice(0, 12);
+  }
+
+  set sortByReleaseDate(value: string) {
+
+    this._selectedReleaseDate = value;
+
+    if (this.filteredProducts.length === 0) {
+      sortProductsByReleaseDate(this.allProductsBySearch, value);
+      this.visibleProducts = this.allProductsBySearch.slice(0, 12);
+    }
+
+    if (this.filteredProducts.length > 0) {
+      sortProductsByReleaseDate(this.filteredProducts, value);
+      this.visibleProducts = this.filteredProducts.slice(0, 12);
+    }
+  }
+
+  resetFilterValues() {
+    this._selectedStatus = '';
+    this._selectedCategory = '';
+    this._selectedReleaseDate = '';
   }
 
   onSearch(): void {
-    this.router.navigate(['search', this._searchDetails]);
+    this.router.navigate(["search", this._searchDetails]);
+    this.query = this._searchDetails;
+    this.sub = this.productService.getProductsByQuery(this._searchDetails).subscribe({
```

Comment:
When you have more than one .subscribe( in a component it makes sense to use 
takeUntil. For more info see:

https://stackoverflow.com/questions/58428791/angular-rxjs-observable-takeuntil-vs-unsubscribe-with-a-subscription

Reply:
I have changed it to 'takeUntil' as you have shown me and it works good. thank you :)